packages/cli/src/commands/deploy/flightcontrol.js

```diff
@@ -55,45 +54,53 @@ export const handler = async ({ side, serve, prisma, dm: dataMigrate }) => {
   const rwjsPaths = getPaths()
 
   const execaConfig = {
+    cwd: rwjsPaths.base,
     shell: true,
     stdio: 'inherit',
-    cwd: rwjsPaths.base,
-    extendEnv: true,
-    cleanup: true,
   }
 
   async function runApiCommands() {
-    if (serve) {
-      console.log('\nStarting api...')
-      await apiServerHandler({
-        port: getConfig().api?.port || 8911,
-        apiRootPath: '/',
-      })
-    } else {
-      console.log('\nBuilding api...')
-      execa.sync('yarn rw build api', execaConfig)
+    if (!serve) {
+      console.log('Building api...')
+      execa.commandSync('yarn rw build api --verbose', execaConfig)
 
-      prisma &&
-        execa.sync(
-          path.join(rwjsPaths.base, 'node_modules/.bin/prisma'),
-          ['migrate', 'deploy', '--schema', `"${rwjsPaths.api.dbSchema}"`],
+      if (prisma) {
+        console.log('Running database migrations...')
+        execa.commandSync(
+          `node_modules/.bin/prisma migrate deploy --schema "${rwjsPaths.api.dbSchema}"`,
           execaConfig
         )
-      dataMigrate && execa.sync('yarn rw dataMigrate up', execaConfig)
+      }
+
+      if (dataMigrate) {
+        console.log('Running data migrations...')
+        execa.commandSync('yarn rw dataMigrate up', execaConfig)
+      }
+
+      return
+    }
+
+    const serverFilePath = path.join(rwjsPaths.api.dist, 'server.js')
+    const hasServerFile = fs.pathExistsSync(serverFilePath)
```

Comment:
Still checking dist specifically here because this command is called after build